v2/controllers/marketplace/meterbase_controller.go

```diff
@@ -1316,11 +1553,115 @@ func labelsForPrometheusOperator(name string) map[string]string {
+		if err != nil {
+			return false, err
+		} else if cmc.UserWorkloadEnabled != nil {
+			return *cmc.UserWorkloadEnabled, nil
+		}
+	}
+	return false, nil
+}
+
+// If this is Openshift 4.6+ check if Monitoring for User Defined Projects is enabled
+// https://docs.openshift.com/container-platform/4.6/monitoring/enabling-monitoring-for-user-defined-projects.html
+func isUserWorkloadMonitoringEnabledOnCluster(cc ClientCommandRunner, infrastructure *config.Infrastructure, reqLogger logr.Logger) (bool, *ExecResult) {
+	userWorkloadMonitoringEnabled := false
+	if infrastructure.HasOpenshift() {
+		if infrastructure.OpenshiftParsedVersion().GTE(utils.ParsedVersion460) {
+
+			enableUserWorkload := false
+			userWorkloadMonitoringConfig := false
+
+			// Check if enableUserWorkload: true in cluster-monitoring-config
+			clusterMonitorConfigMap := &corev1.ConfigMap{}
+			result, err := cc.Do(
+				context.TODO(),
+				GetAction(types.NamespacedName{
+					Namespace: utils.OPENSHIFT_MONITORING_NAMESPACE,
+					Name:      utils.OPENSHIFT_CLUSTER_MONITORING_CONFIGMAP_NAME,
+				}, clusterMonitorConfigMap),
+			)
+			if result.Is(Error) {
+				reqLogger.Error(result.GetError(), "Failed to get cluster-monitoring-config.")
+				return userWorkloadMonitoringEnabled, result
+			} else if result.Is(NotFound) {
+				return userWorkloadMonitoringEnabled, NewExecResult(Continue, reconcile.Result{}, nil)
+			} else if result.Is(Continue) {
+				enableUserWorkload, err = isEnableUserWorkloadConfigMap(clusterMonitorConfigMap)
+				if err != nil {
+					reqLogger.Error(result.GetError(), "Failed to parse cluster-monitoring-config.")
+				}
+			}
+
+			// Check if user-workload-monitoring-config is created
+			userWorkloadMonitoringConfigMap := &corev1.ConfigMap{}
+			result, err = cc.Do(
+				context.TODO(),
+				GetAction(types.NamespacedName{
+					Namespace: utils.OPENSHIFT_USER_WORKLOAD_MONITORING_NAMESPACE,
+					Name:      utils.OPENSHIFT_USER_WORKLOAD_MONITORING_CONFIGMAP_NAME,
+				}, userWorkloadMonitoringConfigMap),
+			)
+			if result.Is(Error) {
+				reqLogger.Error(result.GetError(), "Failed to get user-workload-monitoring-config.")
+				return userWorkloadMonitoringEnabled, result
+			} else if result.Is(NotFound) {
+				return userWorkloadMonitoringEnabled, NewExecResult(Continue, reconcile.Result{}, nil)
+			} else if result.Is(Continue) {
+				userWorkloadMonitoringConfig = true
+			}
+
+			userWorkloadMonitoringEnabled = enableUserWorkload && userWorkloadMonitoringConfig
+			return userWorkloadMonitoringEnabled, result
+		}
+	}
+	return userWorkloadMonitoringEnabled, NewExecResult(Continue, reconcile.Result{}, nil)
+}
+
+func updateUserWorkloadMonitoringEnabledStatus(
+	cc ClientCommandRunner,
+	instance *marketplacev1alpha1.MeterBase,
+	userWorkloadMonitoringEnabledSpec bool,
+	userWorkloadMonitoringEnabledOnCluster bool,
+) (*ExecResult, error) {
+
+	if userWorkloadMonitoringEnabledSpec && userWorkloadMonitoringEnabledOnCluster {
```

Comment:
Just a nit. More a style thing but you can write this without else since you are returning in every function.